script.js

```diff
@@ -0,0 +1,257 @@
+
+////////////1///////////////
+
+let result = prompt ('Ваше имя?', '');
+alert (`Ваше имя ${result}`);
+
+const sayHello = function(name) {
+    if (name === "Mark") {
+        return `Hi, ${name}`;
+    }
+        return `Hello, ${name}`;
+    
+}
+
+
+console.log(sayHello("Oleg"));
+console.log(sayHello("Victor"));
+console.log(sayHello("Mark"));
+
+
+
+const sayHello = (name) => {
+    if (name === "Mark") {
+        return `Hi, ${name}`;
+    }
+        return `Hello, ${name}`;
+    
+}
+
+
+//////////////2///////////////////
+
+console.log(sayHello("Oleg"));
+console.log(sayHello("Victor"));
+console.log(sayHello("Mark"));
+
+
+const hypotenuse = function (n ,m) {
+    if (typeof n === "number" && typeof m === "number") {
+        return ((n**2 + m**2)**(1/2));
+    }
+}
+
+console.log(hypotenuse(3, 4));
+
+const hypotenuse = (n ,m) => {
+    if (typeof n === "number" && typeof m === "number") {
+        return ((n**2 + m**2)**(1/2));
+    }
+}
+
+console.log(hypotenuse(3, 4));
+
+
+
+//////////////3///////////////////
+
+
+
+const minValue = function (value1 ,value2) {
+    if (typeof value1 === "number" && typeof value2 === "number" && value1<value2) {
+        return value1;
+    }
+
+    if (typeof value1 === "number" && typeof value2 === "number" && value1>value2) {
+        return value2;
+    }
```

Comment:
1)Here you are breaking the DRY principe - Don't repeat yourself. You made similar parameters cheching twice.
You can wrap up you code in `if` statement where you check param's types and inside this `if` create another `if` with the logic of finding minimal of them.
2) The best solution for this task is to use `Math.min(value1, value2)`.